### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by Leeoasis**
### Completed Tasks in this PR

- [ ] Remove current_user method in ApplicationController
- [ ] Install and setup devise and ensure that:
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password should be stored in the database.
- Ask for confirmation of email.
- Can reset password.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,8 +1,11 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :current_user
```
Comment (prolajumokeoni) [inline app/controllers/application_controller.rb]: - [ ] Kindly remove the current_user method in ApplicationController, Devise provides a method to do this 👍🏿

File: Gemfile [hunk 2]
```diff
@@ -26,7 +26,6 @@ gem 'stimulus-rails'
 
 # Build JSON APIs with ease [https://github.com/rails/jbuilder]
 gem 'jbuilder'
-gem 'rails-controller-testing'
 # Use Redis adapter to run Action Cable in production
 # gem "redis", "~> 4.0"
```
Comment (prolajumokeoni) [inline Gemfile]: - [ ] Please use the   [getting started link](https://github.com/heartcombo/devise) to setup Devise, the first step from this [guide](https://github.com/heartcombo/devise#getting-started) is to run `bundle add devise`

File: app/models/user.rb [hunk 3]
```diff
@@ -1,12 +1,14 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (prolajumokeoni) [inline app/models/user.rb]: - [ ] Based on the requirements, Please enable `:confirmable` in your  user model 👍🏿

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @Leeoasis and @MNisarAli 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  Descriptive and professional readme 
✔️  Pair-programming 



## Required Changes ♻️

- [ ] Kindly make sure that the Signin and Signup pages follow the design [wireframe
](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md)
- [ ] Kindly make sure that after a user  successfully signs up, they can create a post, I was able to signup after running the devise configuration, but I was unable to create a post
Screenshot_86


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codecaiine — APPROVED** [review]
Hi @Leeoasis @prolajumokeoni @MNisarAli  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good requested change implementation ✔️ 
- Good devise setup ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**